frontend/src/app/ta-application/ta-application.service.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  RxCourseList,
+  RxSectionList
+} from '../academics/academics-admin/rx-academics-admin';
+
+@Injectable({ providedIn: 'root' })
+export class ApplicationsService {
+  private applications: RxApplications = new RxApplications();
+  public applications$: Observable<Application[]> = this.applications.value$;
+
+  private user_application: RxApplication = new RxApplication();
+  public user_application$: Observable<Application | null> =
+    this.user_application.value$;
+
+  private new_uta$ = new BehaviorSubject<boolean>(true);
+  public new_uta: Observable<boolean> = this.new_uta$.asObservable();
+
+  private courses: RxCourseList = new RxCourseList();
+  public courses$: Observable<Course[]> = this.courses.value$;
+
+  private sections: RxSectionList = new RxSectionList();
+  public sections$: Observable<Section[]> = this.sections.value$;
+
+  constructor(protected http: HttpClient) {
+    this.initializeApplicationState();
+  }
+
+  /** Returns a list of all Applications
+   * @returns {Observable<Application[]>}
+   */
+  list(): void {
+    this.http
+      .get<Application[]>('/api/applications')
+      .subscribe((applications) => this.applications.set(applications));
+  }
+
+  initializeApplicationState(): void {
+    this.getApplication().subscribe((application) => {
+      this.user_application.set(application);
+      this.new_uta$.next(!application);
+    });
+  }
+
+  getApplication(): Observable<Application | null> {
+    return this.http
+      .get<Application | null>('/api/applications/user')
+      .pipe(
+        tap((application) => console.log('Fetched application:', application))
+      );
+  }
+
+  submitApplication(
+    application: Omit<Application, 'id'>
+  ): Observable<Application> {
+    return this.user_application.value$.pipe(
+      take(1),
+      switchMap((currentApplication) => {
+        return currentApplication
+          ? this.updateApplication(application)
+          : this.createApplication(application);
+      })
+    );
+  }
+
+  private updateApplication(
+    application: Omit<Application, 'id'>
+  ): Observable<Application> {
+    return this.http
+      .put<Application>(`/api/applications/update`, application)
+      .pipe(
+        tap((updatedApplication) => {
+          this.user_application.set(updatedApplication);
+          console.log('Application updated:', updatedApplication);
+        })
+      );
+  }
+
+  private createApplication(
+    application: Omit<Application, 'id'>
+  ): Observable<Application> {
+    return this.http.post<Application>('/api/applications', application).pipe(
+      tap((newApplication) => {
+        this.user_application.set(newApplication);
+        this.new_uta$.next(false);
+        console.log('Application created:', newApplication);
+      })
+    );
+  }
+
+  deleteApplication(): void {
+    this.http.delete('/api/application/delete');
+  }
+
+  getProfile(): Observable<Profile> {
+    return this.http.get<Profile>('/api/profile');
+  }
```

Comment:
We can use the profile already injected to avoid duplicated code! See the other services for inspiration.